tests/ui/step/SubStepTest.cpp

```diff
@@ -1,97 +1,73 @@
 #include "gtest/gtest.h"
 #include "gmock/gmock.h"
 #include "ui/step/SubSteps.h"
+#include "ui/Factory.h"
 #include "ui/view/ViewMock.h"
-#include "ui/FactoryMock.h"
-#include "ui/ContextMock.h"
 
 using ::testing::Return;
-using ::testing::AtLeast;
-using ::testing::InSequence;
-using ::testing::_;
 
 using namespace ui;
 using namespace ui::step;
 
 class SubStepTest : public ::testing::Test {
 public:
     void SetUp() override {
-        task_ = std::make_shared<Task>();
-        auto reader = std::make_shared<Reader>();
-        auto printer = std::make_shared<Printer>();
-        view_ = std::make_shared<ViewMock>(reader, printer);
+        view_ = std::make_shared<ViewMock>(std::make_shared<Reader>(), std::make_shared<Printer>());
         factory_ = std::make_shared<Factory>(view_);
```

Comment:
Why are you using Mock for states factory `steps` tests, and not using it in `substeps` tests?

Reply:
Fixed